ygen/codegen.go

```diff
@@ -562,7 +595,7 @@ func (cg *YANGCodeGenerator) findRootEntries(structs map[string]*yang.Entry) map
 			// Since we never expect a top-level 'state' or 'config'
 			// container, then it is only such lists that must be
 			// identified.
-			if cg.Config.CompressOCPaths && s.IsList() {
+			if compressPaths && s.IsList() {
```

Comment:
Yes, we need an issue for this. Today, if you turn `CompressOCPaths` on, then we assume that it's an OpenConfig schema. We can't avoid saying that `CompressOCPaths` can *only* deal with OC schemas though - since our path compression is not workable on all schemas. We should, however, make sure it doesn't result in NPEs.